src/Hazelcast.Net.Caching/ProvidedHazelcastCache.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Microsoft.Extensions.Options;
+
+namespace Hazelcast.Caching;
+
+// class to be registered in dependency-injection scenario to properly resolve options
+internal class ProvidedHazelcastCache : HazelcastCache
+{
+    public ProvidedHazelcastCache(IOptions<HazelcastOptions> hazelcastOptions, IOptions<HazelcastCacheOptions> hazelcastCacheOptions)
+        : base(
+            hazelcastOptions?.Value ?? throw new ArgumentNullException(nameof(hazelcastOptions)),
```

Comment:
Are these checks necessary? Because same checks already done in ctor of `HazelcastCache`.

Reply:
We still need to check that e.g. `hazelcastOptions` is not `null` but that can be done more nicely, updating the code